flink-python/src/test/java/org/apache/flink/table/runtime/utils/PassThroughArrowPythonScalarFunctionRunner.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.utils;
+
+import org.apache.flink.python.env.PythonEnvironmentManager;
+import org.apache.flink.table.functions.python.PythonFunctionInfo;
+import org.apache.flink.table.runtime.runners.python.scalar.arrow.ArrowPythonScalarFunctionRunner;
+import org.apache.flink.table.types.logical.RowType;
+
+import org.apache.beam.runners.fnexecution.control.JobBundleFactory;
+import org.apache.beam.sdk.fn.data.FnDataReceiver;
+import org.apache.beam.vendor.grpc.v1p21p0.com.google.protobuf.Struct;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.flink.table.runtime.utils.PythonTestUtils.createMockJobBundleFactory;
+
+/**
+ * An {@link ArrowPythonScalarFunctionRunner} that just return the input elements as the execution results.
+ */
+public class PassThroughArrowPythonScalarFunctionRunner extends ArrowPythonScalarFunctionRunner {
```

Comment:
I'm wondering if we can create a `CommonPassThroughScalarFunctionRunner<IN>` that can be used for both arrow and non-arrow, both row and baserow, so that we don't need so many duplicated classes.

Reply:
I have tried to do that. However I find that it's impossible as these four runners extends different classes. Do you have any suggestions on how to do that? I'm very glad to improve them if it's possible.

Reply:
Good suggestion. As discussed offline, there are currently two PassThrough runners which are used for arrow and non-arrow cases separately.